lib/charms/layer/nagios.py

```diff
@@ -0,0 +1,53 @@
+import os
+import stat
+
+NAGIOS_PLUGINS_DIR = '/usr/lib/nagios/plugins'
+
+
+def install_nagios_plugin(source_path, plugin_name, data=None):
+    """ Install a nagios plugin.
+
+    Args:
+        source_path: Source file path to install
+        plugin_name: Name of the plugin in nagios
+
+    Returns: Full path to installed plugin
+    """
+    if data is None:
+        with open(source_path, "r") as f:
+            data = f.read()
+    dest_path = os.path.join(NAGIOS_PLUGINS_DIR, plugin_name)
+    if os.path.exists(dest_path):
+        # we could complain here, test the files are the same contents, or
+        # just bail. Idempotency is a big deal in Juju, so I'd like to be
+        # ok with being called with the same file multiple times, but we
+        # certainly want to catch the case where multiple layers are using
+        # the same filename for their nagios checks.
+        with open(dest_path, "r") as f:
+            dest = f.read()
+        if dest == data:
+            # same file
+            return dest_path
+        # else:
```

Comment:
Are you intending to handle the 'else' case?

Reply:
Originally I had an else here to complain about duplicate plugin names, but then I realized through my testing that it will be common to rewrite the plugin as configuration changes. This results in being unable to tell the difference, so I removed the error.